lib/access.js

```diff
@@ -1,30 +1,127 @@
 module.exports = access
 
 var assert = require('assert')
+var url = require('url')
 
-function access (uri, params, cb) {
-  assert(typeof uri === 'string', 'must pass registry URI to access')
-  assert(params && typeof params === 'object', 'must pass params to access')
-  assert(typeof cb === 'function', 'muss pass callback to access')
-
-  assert(typeof params.level === 'string', 'must pass level to access')
-  assert(
-    ['public', 'restricted'].indexOf(params.level) !== -1,
-    "access level must be either 'public' or 'restricted'"
-  )
-  assert(
-    params.auth && typeof params.auth === 'object',
-    'must pass auth to access'
-  )
-
-  var body = {
-    access: params.level
-  }
+var subcommands = {}
+
+function access (sub, uri, params, cb) {
+  accessAssertions(sub, uri, params, cb)
+  return subcommands[sub].call(this, uri, params, cb)
+}
 
-  var options = {
+subcommands.public = function (uri, params, cb) {
```

Comment:
Yes, I'm aware. I have to kind of overcompensate to remind myself (also when naming things 'get'). My linter eventually stopped yelling at me about those, so I let myself indulge a little ;)